### ISSUE TITLE
Aspect ratio change with rotation doesn't works with custom shortcut

### ISSUE THREAD

**Issue by pighair** [iina/iina#1168]
**System and IINA version:**

- macOS 10.13
- IINA 0.0.14.1

**Expected behavior:**

Portrait and landscape toggles along "video-rotation +90" or -90 action in keyboard shortcut.

**Actual behavior:**

Window does not change on video rotation defined in keyboard shortcut.

**Steps to reproduce:**

1. Add a keyboard shortcut with action "add video-rotate 90".
2. Play any video which does not have square aspect ratio.
3. Use keyboard shortcut to rotate video.

**How often does this happen?**
Always.

**Comment by pighair** [comment]
I found I can set keyboard shortcuts with multiple commands using ";".
So I tried to assign a keyboard with command below.

set video-rotation 90;set video-aspect "-1"

But iina set video aspect not considering rotation. Please refer following.

Steps:
- Open a 4:3 video
- Change aspect to 16:9
- Use the keyboard shortcut

Result:
- Video rotates 90 degrees and video is resized to 4:3

Expected behavior:
- Video rotates 90 degrees and video is resized to 3:4

**Comment by lhc70000** [comment]
@pighair The multiple command syntax is not supported currently, but now that someone is trying to use it, I'll implement it in the next version.

**Comment by pighair** [comment]
I meant multiple command syntax is working now even thouth UI does not handling it properly and it is not intended. Enhancement I want is to consider rotation status when modifying video’s aspect ratio.
Also I want iina maintains window scale when aspect ratio changes, which I might create another post.

**Comment by gontthias** [comment]
mark

**Comment by low-batt** [comment]
With the fix supplied by @mkyt in PR #3671 (THANKS!) rotation using the methods mentioned by @pighair initially work and then fail:
```text
17:50:53.983 [iina][w] mpv log: [autorotate] error: Video rotation with this format not supported
17:50:54.084 [iina][w] mpv log: [vo/libmpv] warn: Video is flagged as rotated by 359 degrees, but the video output does not support this.
```

The way to get this to completely work is to use this this Lua script: [cycle-video-rotate.lua](https://github.com/VideoPlayerCode/mpv-tools/blob/master/scripts/cycle-video-rotate.lua)

And bind keys to:
```test
script-message Cycle_Video_Rotate 90
script-message Cycle_Video_Rotate -90
```

This is one of the **unofficial 3rd party scripts** listed on the mpv Wiki [User Scripts](
https://github.com/mpv-player/mpv/wiki/User-Scripts) page.

The mpv issue https://github.com/mpv-player/mpv/issues/3869 discusses the problem with `add video-rotate 90` and the need for a script.
